### ISSUE TITLE
Login and Pre-Login UUIDs do not match

### ISSUE THREAD

**Issue by Kikisito** [OKTW-Network/FabricProxy-Lite#1]
**Describe the bug**
When a player joins the server through Velocity, the pre-login event seems to get an offline UUID from the player and the login event an online one, causing mods as LuckPerms to not being able to load players' data correctly. I already tried to report this to LuckPerms's author, more info at https://github.com/lucko/LuckPerms/issues/2873

**Expected behavior**
Both events were expected to have the same UUID.

**FabricProxy-Lite.toml**
Provide you config (without secret):
```
hackOnlineMode = true
secret = "[redacted]"
```

**Versions**
Minecraft version: 1.16.5
FabricProxy-Lite version: 1.0.1
Fabric API version: 0.30.0
Velocity version: 1.1.3

**Server log**
Provide any related logs on the server side:
Log extracted from the issue mentioned above
```
[18:48:07] [Server thread/INFO]: Processing pre-login (sync phase) for 859a0db5-6499-3552-96ea-c8543a67ff75 - Kikisito
[18:48:07] [ForkJoinPool-4-worker-59/INFO]: Processing pre-login (async phase) for 859a0db5-6499-3552-96ea-c8543a67ff75 - Kikisito
[18:48:07] [Server thread/INFO]: Kikisito[/127.0.0.1:62988] logged in with entity id 188 at (253.5, 70.0, -156.5)
[18:48:07] [Server thread/INFO]: Processing login for 31c7be0d-89fc-401e-9d16-bafca8739984 - Kikisito
[18:48:07] [Server thread/WARN]: User 31c7be0d-89fc-401e-9d16-bafca8739984 - Kikisito doesn't currently have data pre-loaded - denying login.
[18:48:07] [Server thread/INFO]: Kikisito joined the game
[18:48:07] [Server thread/INFO]: Kikisito lost connection: [LP] Permissions data for your user was not loaded during the pre-login stage - unable to continue. Please try again later. If you are a server admin, please check the console for any errors.
[18:48:07] [Server thread/INFO]: Kikisito left the game
```

**Comment by james58899** [comment]
Sorry I can't fix it, Fabric-API not allow send and receive packet before QUERY_START fired.

Until I find other ways, Lite version will not work with LuckPerms. 😢 
Use [FabricProxy](https://github.com/OKTW-Network/FabricProxy) instead.

**Comment by james58899** [comment]
I added workaround for it.
Try v1.1.0 and enable `hackEarlySend` in config.

**Comment by Kikisito** [comment]
Thanks for your quick answer. I have just tried the latest version of FabricProxy (1.1.0) and the issue seems to be gone. Thanks for your help!

**Comment by james58899** [comment]
> 
> For some reason this doesn't work for me, I get this error when enabling that option: https://pastebin.com/FviVYQzP
> I am using v1.1.1 FabricProxy-Lite and Velocity 1.1.5

Fixed in v1.1.2.
https://github.com/FabricMC/fabric/pull/1440 changed some internal class cause hack break.

**Comment by ItzRiotz** [comment]
> > 
> > For some reason this doesn't work for me, I get this error when enabling that option: https://pastebin.com/FviVYQzP
> > I am using v1.1.1 FabricProxy-Lite and Velocity 1.1.5
> 
> Fixed in v1.1.2.
> https://github.com/FabricMC/fabric/pull/1440 changed some internal class cause hack break.

Ok, I'll test it soon
